src/internal/data/metrics.go

```diff
@@ -15,3 +22,73 @@ func (cs CollectionStats) IsZero() bool {
 func (cs CollectionStats) String() string {
 	return cs.Details
 }
+
+type KindStats struct {
+	BytesRead     int64
+	ResourceCount int64
+}
+
+type ExportStats struct {
+	// data is kept private so that we can enforce atomic int updates
+	// TODO(meain): Should we be exposing details.ItemType?
+	data map[details.ItemType]KindStats
+}
```

Comment:
This all feels like a recreation of work that's already handled in https://github.com/alcionai/corso/blob/main/src/internal/stats/stats.go.  I've been meaning to revisit our stats gathering and replace them all with this.  Could you plug it in here instead of the bespoke solution?

I'll approve what's here if you want to stick with it, but that does mean I'll unwind it when I get around to the tech debt.

Reply:
I thought of reusing that, but the data that we need is a bit different here. The main difference being that we need separate stats for individual `ItemType`. Also, I am kinda on the fence about exposing exposing `details.ItemType`. Any thoughts on that?

Reply:
I'm a doof 🤦 .  I meant to link this library: https://github.com/alcionai/corso/blob/main/src/pkg/count/count.go

You can get per-type stats by creating local instances for each one.  A nice byproduct is that the stats will naturally centralize into a total at the same time.

Again, I'm happy to handle this later.

Reply:
I'm gonna defer it to a later PR as of now. If you don't get it before me, I'll create the PR to do the switch.